mousestyles/distribution/kolmogorov_test.py

```diff
@@ -0,0 +1,147 @@
+    statistic.
+
+    The Pareto distribution is sometimes known as the Power Law distribution,
+    with the PDF:  b / x**(b + 1) for x >= 1, b > 0.
+    The truncated exponential distribution is the same as the rescaled
+    exponential distribution.
+
+    Parameters
+    ----------
+    distribution: A Scipy Stats Continuous Distribution
+        {stats.pareto, stats.expon, stats.gamma}
+        The distribution to test against. Currently support pareto, expon, and
+        gamma, but any one-sided continuous distribution in Scipy.stats should
+        work.
+    is_plot: boolean
+        If True, will plot the PDF and CDF for visualization
+    verbose: boolean
+        If True, will print out testing result
+
+    Returns
+    -------
+    params: np.ndarray shape (p,)
+        The optimal parameter for the distribution. Optimal in the sense of
+        minimizing K-S statistics.
+    The function also print out the Kolmogorov-Smirnov test result for three
+    cases
+    1. When comparing the empirical distribution against the distribution with
+    parameters estimated with MLE
+    2. When comparing the empirical distribution against the distribution with
+    parameters estimated by explicitely minimizing KS statistics
+    3. When comparing a resample with replacement of the empirical distribution
+    against the Pareto in 2.
+    A p-value > 0.05 means we fail to reject the Null hypothesis that the
+    empirical distribution follow the specified distribution.
+
+    Notes:
+    ------
+    The MLE often does not fit the data very well. We instead minimizing the
+    K-S distance, and obtain a better fit (as seen by the PDF and CDF
+    similarity between sample data and the fit)
+
+    References:
+    -----------
+        1. Kolmogorov-Smirnov test:
+            https://en.wikipedia.org/wiki/Kolmogorov-Smirnov_test
+        2. Pareto Distribution (also known as power law distribution)
+            https://en.wikipedia.org/wiki/Pareto_distribution
+
+    Examples:
+    ---------
+    >>> res = perform_kstest(0, 0, 0, is_plot=False, verbose=False)
+    >>> np.allclose(res, np.array([3.67593246, 0.62795748, 0.37224205]))
+    True
+    """
+
+    x = get_travel_distances(strain, mouse, day)
+    dist_name = distribution.__class__.__name__.split("_")[0]
+    # Fit the parameters by MLE
+    mle_params = distribution.fit(x)
+
+    # Fit the Parameter by explicitely minimize KS-Statistics, and perform
+    # K-S test.
+    def calculate_ks_statistics(params):
+        """This will return the kolmogorov test statistics of the empirical
+        distribution of x and the pareto distribution with parameter params.
+        """
+        return stats.kstest(x, dist_name, args=params)[0]
+
+    if dist_name == "pareto":
+        # A good initial estimate for Pareto parameter to help optimizer
+        # converge to a better parameter
+        init_params = [4.5, 0.5, 0.5]
```

Comment:
there is a problem because this init_params is only good for Pareto distribution. If I have that as a default parameter, and user uses a different distribution (e.g. exponential), then the result will be bad. For other distribution, initialize the search with the mle parameters is better.